sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_MockingReadingEventsTests.cs

```diff
@@ -0,0 +1,290 @@
+                        partitionEvent.Partition.ReadLastEnqueuedEventProperties();
+
+                    // asserting last enqued event properties
+                    Assert.IsNotNull(properties);
+                    Assert.AreEqual(mockPartitionId, partitionEvent.Partition.PartitionId);
+                    Assert.AreEqual(lastSequence, properties.SequenceNumber);
+                    Assert.AreEqual(lastOffset, properties.Offset);
+                    Assert.AreEqual(fakeDate, properties.EnqueuedTime);
+                    Assert.AreEqual(fakeDate, properties.LastReceivedTime);
+                }
+
+                // get event hub properties
+                var eventHubProperties = await consumer.GetEventHubPropertiesAsync(CancellationToken.None);
+
+                // assert event hub properties
+                Assert.IsNotNull(eventHubProperties);
+                Assert.AreEqual(mockEventHubProperties.Name, eventHubProperties.Name);
+                Assert.AreEqual(mockEventHubProperties.PartitionIds, eventHubProperties.PartitionIds);
+                Assert.AreEqual(mockEventHubProperties.CreatedOn, eventHubProperties.CreatedOn);
+
+                // assert partition publishing properties
+                Assert.IsNotNull(mockPartitionPublishingProperties);
+                Assert.IsTrue(mockPartitionPublishingProperties.IsIdempotentPublishingEnabled);
+                Assert.AreEqual(675,mockPartitionPublishingProperties.ProducerGroupId);
+                Assert.AreEqual((short)12,mockPartitionPublishingProperties.OwnerLevel);
+                Assert.AreEqual(4, mockPartitionPublishingProperties.LastPublishedSequenceNumber);
+            }
+            catch (TaskCanceledException)
+            {
+                // This is expected if the cancellation token is
+                // signaled.
+            }
+            finally
+            {
+                await consumer.CloseAsync();
+            }
+
+            mockEventHubConsumerClient.VerifyAll();
+
+            #endregion
+        }
+
+        private IEnumerable<EventData> GenerateBatchData()
+        {
+            var generatedBatchData = new List<EventData>();
+            var properties = new Dictionary<string, object> { { "id", 12 } };
+            var systemProperties = new Dictionary<string, object> { { "custom", "sys-value" } };
+
+            for (int index = 0; index < 5; ++index)
+            {
+                var eventBody = new BinaryData($"This is an example event body #{index}");
+                generatedBatchData.Add(EventHubsModelFactory.EventData(eventBody, properties, systemProperties));
+            }
+
+            return generatedBatchData;
+        }
+
+        private async IAsyncEnumerable<PartitionEvent> GetMockEventsFromPartition(string partitionId,LastEnqueuedEventProperties lastEnqueuedEventProperties = default)
+        {
+            // we are using this as we dont have an async operation in our test
+            await Task.Yield();
```

Comment:
Something like this would make a good stand-alone sample snippet for "how to mock the the async enumerable with last enqueued properties";  we should consider using a simplified form inline with the "read events" snippets, and having this be it's own